drivers/axi_core/axi_dmac/axi_dmac.h

```diff
@@ -81,27 +81,40 @@ enum dma_flags {
 	DMA_LAST = 2
 };
 
+struct axi_dmac_transfer {
+	uint32_t address;
+	uint32_t size;
+};
+
 struct axi_dmac {
 	const char *name;
 	uint32_t base;
 	enum dma_direction direction;
 	uint32_t flags;
+	struct axi_dmac_transfer transfer_lifo[10];
+	volatile uint32_t driver_index;
+	volatile uint32_t user_index;
+	void (*usr_callback)();
 };
 
 struct axi_dmac_init {
 	const char *name;
 	uint32_t base;
 	enum dma_direction direction;
 	uint32_t flags;
+	void (*usr_callback)();
 };
 
 /******************************************************************************/
 /************************ Functions Declarations ******************************/
 /******************************************************************************/
+void axi_dmac_isr(void *instance);
```

Comment:
Why exposing the isr to the header?
Shouldn't be used only in the .c file?

Reply:
The user should explicitly use it, even if they only register it. If I register it by default in .c it inconveniently ties the compilation of axi_dmac to the irq driver.

Reply:
I understand now

Reply:
maybe add default to the name, like: axi_dmac_default_isr